Core/Business/Routing/Router.php

```diff
@@ -0,0 +1,502 @@
+    $controllerInstance = new '.$controllerClass.'($router, $container);
+';
+                foreach ($initTraits as $initTrait) {
+                    $phpCode .= '
+    $actionAllowed = $actionAllowed & $controllerInstance->'.$initTrait.'($request, $response);';
+                }
+                foreach ($beforeActionTraits as $beforeActionTrait) {
+                    $phpCode .= '
+    $actionAllowed = $actionAllowed & $controllerInstance->'.$beforeActionTrait.'($request, $response);';
+                }
+                if ($controllerResolverTrait) {
+                    $phpCode .= '
+
+    if ($actionAllowed) {
+        $controllerResolver = $controllerInstance->'.$controllerResolverTrait.'($request, $response);
+        if ($controllerResolver) {
+            $responseFormat = $controllerResolver($controllerInstance, $method);
+            if ($responseFormat) {
+                $response->setResponseFormat($responseFormat);
+            }
+        } else {
+            throw new DevelopmentErrorException(\'The controller uses a Trait controllerResolver that failed to return a controllerResolver!\');
+        }
+    }
+';
+                } else {
+                    $phpCode .= '
+
+    if ($actionAllowed) {
+        $responseFormat = $controllerInstance->'.$controllerMethod.'($request, $response);
+        if ($responseFormat) {
+            $response->setResponseFormat($responseFormat);
+        }
+    }
+';
+                }
+                foreach ($afterActionTraits as $afterActionTrait) {
+                    $phpCode .= '
+    $actionAllowed = $actionAllowed & $controllerInstance->'.$afterActionTrait.'($request, $response);';
+                }
+                foreach ($closeActionTraits as $closeActionTrait) {
+                    $phpCode .= '
+    $actionAllowed = $actionAllowed & $controllerInstance->'.$closeActionTrait.'($request, $response);';
+                }
+                
+                $phpCode .= '
+
+    if ($actionAllowed && ($responseFormat = $response->getResponseFormat())) {
+        list($encapsulation, $format) = explode(\'_\', $responseFormat);
+        if ($format) {
+            $controllerInstance->formatResponse($format, $response);
+        }
+        if ($encapsulation) {
+            $controllerInstance->encapsulateResponse($encapsulation, $response);
+        }
+';
+                if ($returnView) {
+                    $phpCode .= '
+        // Do not use send (no output buffer tricks)
+        return $response;
+    }
+
+    if (!$actionAllowed) {
+        throw new \Core_Foundation_Exception_Exception(\'Action forbidden.\');
+    }
+}
+'; // Raw php code inside a string, do not indent please.
+                } else {
+                    $phpCode .= '
+        // Send response to output buffer
+        $response->send();
+    }
+
+    if (!$actionAllowed) {
+        $router->redirectToForbidden();
+    }
+    return true;
+}
+'; // Raw php code inside a string, do not indent please.
```

Comment:
I think this is on of the part to be modified next but I'm still a comment on this. I think we can do better